types/testingutils/beacon_node_versioned.go

```diff
@@ -174,8 +242,64 @@ var TestingInvalidDutySlotV = func(version spec.DataVersion) phase0.Slot {
 	switch version {
 	case spec.DataVersionBellatrix:
 		return TestingDutySlotBellatrixInvalid
+	case spec.DataVersionCapella:
+		return TestingDutySlotCapellaInvalid
 
 	default:
 		panic("unsupported version")
 	}
 }
+
+var TestingBeaconBlockCapella = func() *capella.BeaconBlock {
+	var res capella.BeaconBlock
+	err := json.Unmarshal(capellaBlock, &res)
+	if err != nil {
+		panic(err)
+	}
+	// using TestingDutySlotCapella to keep the consistency with TestingProposerDutyV Capella slot
+	res.Slot = TestingDutySlotCapella
+	return &res
+}()
+
+var TestingBlindedBeaconBlockCapella = func() *apiv1capella.BlindedBeaconBlock {
```

Comment:
why is the full block taken from a json and the versioned one generated here?

Reply:
I'm unsure about the issue with the JSON format. Firstly, it improves readability, and secondly, it's utilized by a reliable package (go-eth2-client) that the entire SSV spec relies on.

Apart from that, I believe the question might contain a typo. Alon's question states:
> "why is the full block taken from a JSON and the versioned one generated here?"

I assume he meant "blinded" instead of "versioned." @alonmuroch, please correct me if I'm mistaken.

Addressing the question:
- Why not? We employ the same logic to construct the blinded block from the full block. The only difference is that the Bellatrix full block was manually built, whereas Capella's block is derived from the JSON.

By the way, the [SSZTransactions](https://github.com/olegshmuelov/ssv-spec/blob/8a9e52945bf721763e7fe7f537f9b2a2d8cdf401/types/ssz.go#L53) were created for a specific purpose:
1. to build the blinded block
2. verify that the hash root of the full data block matches the blinded one. You can refer to the [related test](https://github.com/olegshmuelov/ssv-spec/blob/8a9e52945bf721763e7fe7f537f9b2a2d8cdf401/types/testingutils/beacon_node_test.go#L10) for more information.

That's why we've maintained the same design and introduced [SSZWithdrawals](https://github.com/olegshmuelov/ssv-spec/blob/8a9e52945bf721763e7fe7f537f9b2a2d8cdf401/types/ssz.go#L87) for Capella, to ensure that the hash root of the full data Capella block matches the blinded one. With that being said, we can deprecate SSZWithdrawals and use the blinded Capella block in JSON format from go-eth2-client, while still retaining the test to verify that their hash roots are identical.

Reply:
In a private conversation @olegshmuelov convinced me.
It is better to have json only copy-paste from `go-eth-client` and get rid of some custom `SSZ` functions we have